test/list_test.go

```diff
@@ -178,32 +178,95 @@ func TestList(t *testing.T) {
 	}
 }
 
-// BenchmarkList is a benchmark for the Get operation.
 func BenchmarkList(b *testing.B) {
-	for _, backendType := range []string{endpoint.SQLiteBackend, endpoint.DQLiteBackend} {
-		b.Run(backendType, func(b *testing.B) {
-			b.StopTimer()
-			g := NewWithT(b)
+	setup := func(ctx context.Context, tx *sql.Tx, payloadSize, n int) error {
+		if err := insertMany(ctx, tx, "key", payloadSize, n); err != nil {
+			return err
+		}
+		if err := updateMany(ctx, tx, "key", payloadSize, n/2); err != nil {
+			return err
+		}
+		if err := deleteMany(ctx, tx, "key", n/2); err != nil {
+			return err
+		}
+		return nil
+	}
+	backends := []string{endpoint.SQLiteBackend, endpoint.DQLiteBackend}
+	for _, backendType := range backends {
+		payloads := []struct {
+			name string
+			size int
+		}{{
+			name: "tiny",
+			size: 100,
+		}, {
+			name: "page-fitting",
+			size: 1000,
+		}, {
+			name: "page-overflowing",
```

Comment:
Somehow these read like adjectives, but I think these should instead be `fits-in-page` and `overflows-page`